pytorch_lightning/callbacks/timer.py

```diff
@@ -0,0 +1,133 @@
+    if the given time limit is reached.
+
+    Args:
+        duration: A string in the format DD:HH:MM:SS (days, hours, minutes seconds), or a :class:`datetime.timedelta`,
+            or a dict containing key-value compatible with :class:`~datetime.timedelta`.
+        interval: Determines if the interruption happens on epoch level or mid-epoch.
+            Can be either `epoch` or `step`.
+        verbose: Set this to ``False`` to suppress logging messages.
+
+    Raises:
+        MisconfigurationException:
+            If ``interval`` is not one of the supported choices.
+
+    Example::
+        from pytorch_lightning import Trainer
+        from pytorch_lightning.callbacks import Timer
+
+        # stop training after 12 hours
+        timer = Timer(duration="00:12:00:00")
+
+        # or provide a datetime.timedelta
+        from datetime import timedelta
+        timer = Timer(duration=timedelta(weeks=1))
+
+        # or provide a dictionary
+        timer = Timer(duration=dict(weeks=4, days=2))
+
+        # force training to stop after given time limit
+        trainer = Trainer(callbacks=[timer])
+    """
+
+    def __init__(
+        self,
+        duration: Union[str, timedelta, Dict[str, int]],
+        interval: str = Interval.step,
+        verbose: bool = True,
+    ):
+        super().__init__()
+        if isinstance(duration, str):
+            dhms = duration.strip().split(":")
+            dhms = [int(i) for i in dhms]
+            duration = timedelta(days=dhms[0], hours=dhms[1], minutes=dhms[2], seconds=dhms[3])
+        if isinstance(duration, dict):
+            duration = timedelta(**duration)
+        if interval not in set(Interval):
+            raise MisconfigurationException(
+                f"Unsupported parameter value `Timer(interval={interval})`. Possible choices are:"
+                f" {', '.join(set(Interval))}"
+            )
+        self._duration = duration
+        self._interval = interval
+        self._verbose = verbose
+        self._start_time = None
+        self._offset = timedelta()
+
+    @property
+    def start_time(self) -> datetime:
+        return self._start_time
+
+    @property
+    def time_elapsed(self) -> timedelta:
+        if self._start_time is None:
+            return self._offset
+        return datetime.now() - self._start_time + self._offset
+
+    @property
+    def time_remaining(self) -> timedelta:
+        return self._duration - self.time_elapsed
+
+    def on_train_start(self, trainer, *args, **kwargs) -> None:
+        self._start_time = datetime.now()
+
+    def on_train_batch_end(self, trainer, *args, **kwargs) -> None:
+        if self._interval != Interval.step:
+            return
+        self._check_time_remaining(trainer)
+
+    def on_train_epoch_end(self, trainer, *args, **kwargs) -> None:
+        if self._interval != Interval.epoch:
+            return
+        self._check_time_remaining(trainer)
```

Comment:
in case people extend this callback, *args/**kwargs makes these callbacks harder to provide typehints for

Reply:
I'm not sure I understand, because I am a typehint noob.
The arguments are unused, therefore they can be of type Any. 
If I did specify all the args, linter will complain they are unused.

Reply:
I switched everything to monotonic, but note now `time_elapsed()` etc returns seconds. 
Hope that's fine.